src/Module/Api/Mastodon/Statuses/Favourite.php

```diff
@@ -42,13 +42,16 @@ protected function post(array $request = [])
 			DI::mstdnError()->UnprocessableEntity();
 		}
 
-		$item = Post::selectOriginalForUser($uid, ['id'], ['uri-id' => $this->parameters['id'], 'uid' => [$uid, 0]]);
+		$item = Post::selectOriginalForUser($uid, ['id', 'uri-id'], ['uri-id' => $this->parameters['id'], 'uid' => [$uid, 0]]);
 		if (!DBA::isResult($item)) {
 			DI::mstdnError()->RecordNotFound();
 		}
 
 		Item::performActivity($item['id'], 'like', $uid);
 
-		System::jsonExit(DI::mstdnStatus()->createFromUriId($this->parameters['id'], $uid, self::appSupportsQuotes())->toArray());
+		// @TODO Remove once mstdnStatus()->createFromUriId is fixed
+		$isReblog = $item['uri-id'] != $this->parameters['id'];
```

Comment:
I'm unsure about this change as well.

Reply:
Same as above, trying to address the problem I identified here: https://github.com/friendica/friendica/pull/13175#issuecomment-1591079883